src/util/health-check.sh

```diff
@@ -14,12 +14,13 @@
 
 TIMEOUT=10
 
-read -t ${TIMEOUT} -r input
+# prefer read with timeout (bash, busybox sh) or fall back to POSIX read (dash)
+read -t ${TIMEOUT} -r INPUT 2>/dev/null || read -r INPUT
```

Comment:
Maybe using trap and get rid of `read -t` overall is nicer. So use something like this:

```
trap "exit 0" QUIT
{
	sleep $timeout
	kill -3 $$ 2>/dev/null
} &
```
and just run `read -r INPUT`. This solves the problem that dash does not feature read timeout, which i think of as an important feature for this.

Reply:
Thank you for the suggestion - this lets us use the timeout for all shells! I updated the branch and added this change.